### PR TITLE
Use docker-storage-setup instead of manual setup

### PR DESCRIPTION
**Pull request by sjug**
Instead of setting up our docker storage manually, this uses docker-storage-setup. Needs a bit more testing with devicemapper before we can merge.

### CODE REVIEW

File: image_provisioner/playbooks/roles/docker-config/templates/etc/sysconfig/docker-storage-setup.j2 [hunk 1]
```diff
@@ -1,2 +1,12 @@
-VG={{ docker_storage_vgname }}
-STORAGE_DRIVER={{ docker_storage_driver }}
+DEVS="{{ docker_storage_device }}"
+VG="{{ docker_storage_vgname }}"
+
+{% set storagedriver = docker_storage_driver | default('devicemapper') %}
```
Comment (sjug) [inline image_provisioner/playbooks/roles/docker-config/templates/etc/sysconfig/docker-storage-setup.j2]: Not really necessary to create new variable name, just adds default if unset

File: image_provisioner/playbooks/roles/docker-config/templates/etc/sysconfig/docker-storage-setup.j2 [hunk 2]
```diff
@@ -1,2 +1,12 @@
...
+STORAGE_DRIVER="{{ storagedriver }}"
+{% if storagedriver == 'devicemapper' %}
+DATA_SIZE=95%VG
```
Comment (sjug) [inline image_provisioner/playbooks/roles/docker-config/templates/etc/sysconfig/docker-storage-setup.j2]: Could also move to `100%FREE`

File: image_provisioner/playbooks/roles/docker-config/templates/etc/sysconfig/docker-storage-setup.j2 [hunk 3]
```diff
@@ -1,2 +1,12 @@
-VG={{ docker_storage_vgname }}
-STORAGE_DRIVER={{ docker_storage_driver }}
+DEVS="{{ docker_storage_device }}"
+VG="{{ docker_storage_vgname }}"
```
Comment (sjug) [inline image_provisioner/playbooks/roles/docker-config/templates/etc/sysconfig/docker-storage-setup.j2]: I don't think this needs to be exposed in the inventory

File: image_provisioner/playbooks/roles/docker-config/tasks/main.yaml [hunk 4]
```diff
@@ -57,20 +45,16 @@
...
-     ignore_errors: true
-  
-   - name: exit if kernel version is outdated
-     fail: 
-       msg: "please update the kernel to 3.10.0-517 or later"
-     when: vercmp_out.rc == 12
+     failed_when: vercmp_out.rc == 12
   when: docker_storage_driver == "overlay2" or update_kernel == "true"
 
+# We can remove this but need to test without it
```
Comment (sjug) [inline image_provisioner/playbooks/roles/docker-config/tasks/main.yaml]: Still need to test without this block for devicemapper

**Comment by sjug** [comment]
May need some changes for atomic host support, will test.

File: image_provisioner/playbooks/roles/docker-config/templates/etc/sysconfig/docker-storage-setup.j2 [hunk 5]
```diff
@@ -1,2 +1,15 @@
-VG={{ docker_storage_vgname }}
-STORAGE_DRIVER={{ docker_storage_driver }}
+{% if atomic == False %}
```
Comment (sjug) [inline image_provisioner/playbooks/roles/docker-config/templates/etc/sysconfig/docker-storage-setup.j2]: Need better condition with default

**Comment by sjug** [comment]
Will finalize testing today, it seems that RHEL doesn't like the `CONTAINER_ROOT_*` params.

**Comment by chaitanyaenr** [comment]
@sjug it's working on atomic for both devicemapper and overlay2.

**Comment by sjug** [comment]
Thanks @chaitanyaenr, verified working on RHEL with both storage drivers (dm & overlay2) as well.

**Pull request merged**